pkg/agent/core/ngt/service/ngt.go

```diff
@@ -215,6 +198,33 @@ func New(cfg *config.NGT, opts ...Option) (nn NGT, err error) {
 	return n, nil
 }
 
+func (n *ngt) prepareFolders() (err error) {
+	if n.enableCopyOnWrite && !n.inMem && len(n.path) != 0 {
+		sep := string(os.PathSeparator)
+		n.path, err = filepath.Abs(strings.ReplaceAll(n.path, sep+sep, sep))
+		if err != nil {
+			// TODO: maybe we should return error here?
+			log.Warn(err)
```

Comment:
@kpango 
What do you think?

Reply:
Our implementation before refactoring did not handle this error but just output it to the log and then continue processing.

There are 3 reasons for previous impl.
1. Abs fails in some cases such as os.Getwd (unix) and syscall.FullPath (windows), but it is rare for these to fail.
2. The motivation for using Abs Path is to eliminate uncertainty caused by relative paths.
3. Even if Abs Path fails, it is possible to start with relative path by using n.path.

However, the previous implementation directly rewrites n.path by filepath.Abs, so it should have been implemented as follows.
```
absPath, err := filepath.Abs(strings.ReplaceAll(n.path, sep+sep, sep))
if err != nil {
        log.Warn(err)
} else if absPath != "" {
        n.path = absPath
}
```

Reply:
> There are 3 reasons for previous impl.

Understood. Thanks.

> However, the previous implementation directly rewrites n.path by filepath.Abs, so it should be implemented as follows.

I think there are no side effects, so I think it's fine, but are there any cases where `file.Exists(absPath)` actually becomes false after `filepath.Abs` successes?

Reply:
any way implemented that way
[db561f2](https://github.com/vdaas/vald/pull/2040/commits/db561f21f9a32560ff43ba4645d030e90dbe757d)

Reply:
> I think there are no side effects so it should be fine, but are there any cases where file.Exists(absPath) actually becomes false after filepath.Abs successes?
I checked go's internal code, there is no cases like that.
Can you please remove absPath checking with file.Exists?

Reply:
👍 
Also added some log msg if that's ok.
[5812658](https://github.com/vdaas/vald/pull/2040/commits/58126583e24d2c481340da257137c756cf192d91)